ocrd_validators/ocrd_validators/page_validator.py

```diff
@@ -169,40 +252,45 @@ def validate_consistency(node, page_textequiv_consistency, page_textequiv_strate
             parent = node
         if parent:
             parent_points = parent.get_Coords().points
-            node_poly = Polygon(polygon_from_points(parent_points))
-            reason = ''
-            if not node_poly.is_valid:
-                reason = explain_validity(node_poly)
-            elif node_poly.is_empty:
-                reason = 'is empty'
-            elif node_poly.bounds[0] < 0 or node_poly.bounds[1] < 0:
-                reason = 'is negative'
-            elif node_poly.length < 4:
-                reason = 'has too few points'
-            if reason:
-                report.add_error(CoordinateValidityError(tag, node_id, file_id, parent_points, reason))
+            node_poly = make_poly(polygon_from_points(parent_points))
+            if not isinstance(node_poly, Polygon):
+                report.add_error(CoordinateValidityError(tag, node_id, file_id,
+                                                         parent_points, node_poly))
                 log.debug("Invalid coords of %s %s", tag, node_id)
                 consistent = False
+                node_poly = None # don't use in further comparisons
         else:
             node_poly = None
+    for class_, getterLO, getterRD in _ORDER[1:]:
+        if isinstance(node, class_):
+            if getterLO:
+                textLineOrder = getattr(node, getterLO)()
+            if getterRD:
+                readingDirection = getattr(node, getterRD)()
     for class_, getter, concatenate_with in _HIERARCHY:
         if not isinstance(node, class_):
             continue
         children = getattr(node, getter)()
+        if (getter == 'get_TextRegion' and children and
+            all(child.id in readingOrder for child in children) and
+            isinstance(readingOrder[children[0].id], (OrderedGroupType, OrderedGroupIndexedType))):
```

Comment:
Not sure I understand what I intended to do here anymore.

I now think this should read:
```suggestion
            node.id in readingOrder and
            isinstance(readingOrder[node.id], (OrderedGroupType, OrderedGroupIndexedType))):
```